### PR TITLE
many: cleanup/update rest.md; improve auth errors

### PR DESCRIPTION
**Pull request by chipaca**

### CODE REVIEW

File: store/errors.go [hunk 1]
```diff
@@ -60,3 +61,17 @@ type ErrDownload struct {
...
+type ErrInvalidAuthData map[string][]string
+
+func (e ErrInvalidAuthData) Error() string {
+	var es []string
+	for _, v := range e {
+		es = append(es, v...)
+	}
+	// XXX: confirm with server people that extra args are all
+	//      full sentences (with periods and capitalization)
+	//      (empirically this checks out)
```
Comment (mvo5) [inline store/errors.go]: `this checks out` is a term I had to look up ;) and learned something new.

**Comment by mvo5** [comment]
Looks good! Thanks for doing this!

File: docs/rest.md [hunk 2]
```diff
@@ -138,8 +142,12 @@ Reserved for human-readable content describing the service.
...
 ```javascript
 {
- "flavor": "core",
  "series": "16",
+ "version": "2.0.17",
+ "os-release": {
+   "id": "ubuntu",
+   "version-id": "17.04",
+ },
  "store": "store-id"          // only if not default
```
Comment (robert-ancell) [inline docs/rest.md]: I couldn't see "store" being set anywhere in the snapd code (could be wrong here).
This is missing "on-classic".

Comment (chipaca) [inline docs/rest.md]: you are correct on both points. Fixed.

**Pull request merged**